### ISSUE TITLE
NullPointerException - BaseSeries$1.<init>(BaseSeries.java:183)

### ISSUE THREAD

**Issue by agilob** [jjoe64/GraphView#397]
I get this exception after a few minutes (device dependent) of running code at the bottom.

```
java.lang.NullPointerException
at com.jjoe64.graphview.series.BaseSeries$1.<init>(BaseSeries.java:183)
at com.jjoe64.graphview.series.BaseSeries.getValues(BaseSeries.java:170)
at com.jjoe64.graphview.series.PointsGraphSeries.draw(PointsGraphSeries.java:168)
at com.jjoe64.graphview.GraphView.onDraw(GraphView.java:298)
at android.view.View.draw(View.java:13554)
at android.view.View.getDisplayList(View.java:12505)
at android.view.View.getDisplayList(View.java:12549)
at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:2911)
at android.view.View.getDisplayList(View.java:12441)
at android.view.View.getDisplayList(View.java:12549)
at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:2911)
at android.view.View.getDisplayList(View.java:12441)
at android.view.View.getDisplayList(View.java:12549)
at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:2911)
at android.view.View.getDisplayList(View.java:12441)
at android.view.View.getDisplayList(View.java:12549)
at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:2911)
at android.view.View.getDisplayList(View.java:12441)
at android.view.View.getDisplayList(View.java:12549)
at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:2911)
at android.view.View.getDisplayList(View.java:12441)
at android.view.View.getDisplayList(View.java:12549)
at android.view.HardwareRenderer$GlRenderer.draw(HardwareRenderer.java:1165)
at android.view.ViewRootImpl.draw(ViewRootImpl.java:2175)
at android.view.ViewRootImpl.performDraw(ViewRootImpl.java:2047)
at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:1853)
at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1013)
at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:4245)
at android.view.Choreographer$CallbackRecord.run(Choreographer.java:725)
at android.view.Choreographer.doCallbacks(Choreographer.java:555)
at android.view.Choreographer.doFrame(Choreographer.java:525)
at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:711)
at android.os.Handler.handleCallback(Handler.java:615)
at android.os.Handler.dispatchMessage(Handler.java:92)
at android.os.Looper.loop(Looper.java:153)
at android.app.ActivityThread.main(ActivityThread.java:5071)
at java.lang.reflect.Method.invokeNative(Native Method)
at java.lang.reflect.Method.invoke(Method.java:511)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:790)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:557)
at de.robv.android.xposed.XposedBridge.main(XposedBridge.java:132)
at dalvik.system.NativeStart.main(Native Method)
```

Setup:
```
graph = (GraphView) findViewById(R.id.graph);

        graph.getViewport().setXAxisBoundsManual(true);
        graph.getViewport().setYAxisBoundsManual(true);

        graph.getGridLabelRenderer().setVerticalAxisTitle(this.getString(R.string.RSSI_dbm));
        graph.getGridLabelRenderer().setHighlightZeroLines(true);
        graph.getGridLabelRenderer().setHorizontalLabelsVisible(false);
        graph.getGridLabelRenderer().setGridStyle(GridLabelRenderer.GridStyle.HORIZONTAL);
        graph.getGridLabelRenderer().setNumVerticalLabels(3);

        graph.getViewport().setMaxX(60);
        graph.getViewport().setMinX(0);

        graph.getViewport().setMaxY(-50);
        graph.getViewport().setMinY(-100);
```

```
private CellStrengthReceiver() {
            super();

            mMeasurements = new LineGraphSeries<>();
            mMeasurements.setColor(Color.GREEN);
            graph.addSeries(mMeasurements);

            highlight = new PointsGraphSeries<>();
            highlight.setSize(5f);
            highlight.setColor(Color.RED);
            graph.addSeries(highlight);
        }

        @Override
        public void onSignalStrengthsChanged(SignalStrength signalStrength) {
            super.onSignalStrengthsChanged(signalStrength);

            mCurrentLevel = (signalStrength.getGsmSignalStrength() * 2 - 113);

            if(Math.abs(mPreviousLevel-mCurrentLevel) > 9){
                highlight.appendData(new DataPoint(graph2LastXValue, mCurrentLevel), true, 60);
            }

            // 99 * 2 - 113 = 85
            if(mCurrentLevel != 85) {
                mMeasurements.appendData(new DataPoint(graph2LastXValue, mCurrentLevel), true, 60);
            } else {
                mMeasurements.appendData(new DataPoint(graph2LastXValue, 0), true, 60);
            }

            ccSignalTextView.setText(String.valueOf(mCurrentLevel));
            mPreviousLevel = mCurrentLevel;
            graph2LastXValue += 1d;
        }
    }
```

**Comment by jjoe64** [comment]
fixed

**Comment by GeorgeSia** [comment]
Hi, I have the same issue. Could you please tell me how to fix it?

**Comment by jjoe64** [comment]
should be fixed in current development state. will be released in version 4.1 in 1-2 weeks

**Comment by agilob** [comment]
cheers!
